### PR TITLE
BUG: fixed _stats of t-distribution in scipy.stats

### LINKED ISSUE #8786: erroneous moment values for t-distribution
<<Please describe the issue in detail here, and for bug reports fill in the fields below.>>

Moments of the t-distribution return incorrect values.  See: https://en.wikipedia.org/wiki/Student%27s_t-distribution
- for df <= 1, the mean should be undefined
- for df < 1, the variance should be undefined 

Example:
```
from scipy import stats
X = stats.t(df=0.8)
X.mean() # returns 0
X.var() # returns inf
```

### Scipy/Numpy/Python version information:
`1.0.0 1.14.0 sys.version_info(major=3, minor=6, micro=4, releaselevel='final', serial=0)`

### PR DESCRIPTION
**Pull request by chrisb83**
Closes #8786

I tried to improve the doc string as well:
- make reference to `scipy.special.gamma`
- this function should be written as \Gamma in Latex to be aligned to the docstring of the gamma function (I noted that lower case (\gamma) is used in other distributions as well, maybe align in separate PR?)

Still need to add test cases.

### CODE REVIEW

File: scipy/stats/_continuous_distns.py [hunk 1]
```diff
@@ -4662,12 +4662,13 @@ class t_gen(rv_continuous):
 
     .. math::
 
-        f(x, df) = \frac{\gamma((df+1)/2)}
-                        {\sqrt{\pi*df} \gamma(df/2) (1+x^2/df)^{(df+1)/2}}
+        f(x; \text{df}) = \frac{\Gamma((\text{df}+1)/2)}
+                        {\sqrt{\pi*\text{df}} \Gamma(\text{df}/2)}
+                    (1+x^2/\text{df})^{-(\text{df}+1)/2}
```
Comment (ev-br) [inline scipy/stats/_continuous_distns.py]: Possibly rename the parameter to \nu in the formula and mention in the text that it corresponds to `df` in code?

**Comment by ev-br** [comment]
One DOC comment, otherwise LGTM.

**Comment by ev-br** [comment]
> I noted that lower case (\gamma) is used in other distributions as well, maybe align in separate PR?

Indeed, `scipy.special.gamma` should be spelled as an uppercase $\Gamma$ everywhere. Best send a separate PR, so that we can merge this one once CI finishes?

**Comment by chrisb83** [comment]
Hm, this is tricky, the test `check_kurt_expect` expects that `k` is `nan` instead of `inf`. 
I think `inf` should be ok since both the mean and variance are finite here, but the fourth moment is infinite. This is also what Wikipedia says...
```
distfn = <scipy.stats._continuous_distns.t_gen object at 0x7fe301206cf8>
arg = (2.7433514990818093,), m = array(0.0), v = array(3.6905172081719186)
k = array(inf), msg = 't'
    def check_kurt_expect(distfn, arg, m, v, k, msg):
        if np.isfinite(k):
            m4e = distfn.expect(lambda x: np.power(x-m, 4), arg)
            npt.assert_allclose(m4e, (k + 3.) * np.power(v, 2), atol=1e-5, rtol=1e-5,
                    err_msg=msg + ' - kurtosis')
        else:
>           npt.assert_(np.isnan(k))
E           AssertionError
```

**Comment by chrisb83** [comment]
I slightly modified the test so that `k == np.inf` works and not only `nan` in case `k` is not finite. I think this is should be OK, maybe one could also try to detect that the integral diverges. (If one computes `m4e = stats.t(2.7433514990818093).expect(lambda x: np.power(x, 4)))`, there is a warning that integral is probably divergent and `m4e == -24.1699...` which does not make sense).

**Comment by chrisb83** [comment]
@ev-br: If you have time, can you take a look please? Thanks

I will create a separate PR to change \gamma to \Gamma for the other distributions.

File: scipy/stats/_continuous_distns.py [hunk 2]
```diff
@@ -4662,18 +4662,23 @@ class t_gen(rv_continuous):
 
     .. math::
 
-        f(x, df) = \frac{\gamma((df+1)/2)}
-                        {\sqrt{\pi*df} \gamma(df/2) (1+x^2/df)^{(df+1)/2}}
+        f(x; \nu) = \frac{\Gamma((\nu+1)/2)}
+                        {\sqrt{\pi*\nu} \Gamma(\nu)}
```
Comment (ev-br) [inline scipy/stats/_continuous_distns.py]: Nitpick: asterisk not needed in the math mode.

**Review by ev-br — COMMENTED** [review]
LGTM modulo a trivial nitpick, which can be fixed separately in a sequel PR. Merging, thanks @chrisb83. Keep them coming :-)

**Pull request merged**